torch/_decomp/decompositions.py

```diff
@@ -4343,6 +4343,57 @@ def _weight_norm_interface(x, y, dim):
     norm = x.norm(2, keep_dim, keepdim=True)
     return x * (y / norm), norm
 
+@register_decomposition(aten.isin)
+@out_wrapper()
+def isin(elements, test_elements, *, assume_unique=False, invert=False):
+    if test_elements.numel() < 10.0*torch.pow(torch.tensor(elements.numel(), dtype=torch.float32), torch.tensor(0.145)):
```

Comment:
Test failures are due to the tensor ops here, it should probably be using a python expression
```suggestion
    if test_elements.numel() < 10.0*pow(elements.numel(), 0.145):
```

Not sure if we support symbolic pow for arbitrary exponents currently though.

I'll also not that this doesn't play too nicely with dynamic shapes, since evaluating the comparison creates a guard that will force the graph to be recompiled if the sizes cross this threshold. It might be better to make a decision based on the hint value `symbolic_shapes.hint_int(tensor.numel())`. This would mean the choice of algorithm is baked in based on the examples values used for the first compilation.